language/compiler/ir-to-bytecode/syntax/src/lexer.rs

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::syntax::ParseError;
+extern crate regex as __regex;
+use std::{convert::TryInto, fmt};
+
+// FIXME: This is a simplified version of the lexer generated by lalrpop.
+// It should be replaced with something sane.
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum Tok {
+    EOF = -1,
```

Comment:
why -1 versus 0?

Reply:
It's so horrible that I hesitate to tell you.... These enum values should not even be specified. The reason is that they must correlate to indices into the array of regexps generated by lalrpop, and lalrpop doesn't use a special token for EOF. All of this code needs to die (but I still want to do it in a separate step).